src/mscorlib/shared/System/Collections/Generic/Dictionary.cs

```diff
@@ -759,6 +759,22 @@ IEnumerator IEnumerable.GetEnumerator()
             return new Enumerator(this, Enumerator.KeyValuePair);
         }
 
+        /// <summary>
+        /// Ensures that the dictionary can hold up to 'capacity' entries without any further expansion of its backing storage
+        /// </summary>
+        public int EnsureCapacity(int capacity)
+        {
+            if (capacity < 0)
+                ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.capacity);
+            if (_entries != null && _entries.Length >= capacity)
+                return _entries.Length;
+            if (_buckets == null)
+                Initialize(0);
```

Comment:
I think this can be an if/else and use `capacity` as the value for `Initialize` e.g. you don't need to `Resize` after initializing